From 4eebe76e9671afcdf8f468a91c9eb27ddba30715 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?P=C3=A1draig=20Brady?= <pbrady@redhat.com>
Date: Sat, 8 Sep 2012 02:43:01 +0100
Subject: [PATCH 15/21] fix unmounting of LXC containers in the presence of
 symlinks

When looking up the mount dir in /proc/mounts we need
to canonicalize the path so that symlinks are resolved

Fixes bug: 1046313
Change-Id: Ib2bb8253e9cdfeda3c6ba13e56a0f907518ff558
Cherry-picked: b35b34a65a6ddcfee524304cc548b4dfe1fc6cd5
---
 nova/virt/disk/api.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/nova/virt/disk/api.py b/nova/virt/disk/api.py
index e28a76c..dc78b2b 100644
--- a/nova/virt/disk/api.py
+++ b/nova/virt/disk/api.py
@@ -176,6 +176,7 @@ class _DiskImage(object):
     @staticmethod
     def _device_for_path(path):
         device = None
+        path = os.path.realpath(path)
         with open("/proc/mounts", 'r') as ifp:
             for line in ifp:
                 fields = line.split()
-- 
1.7.11.4

